src/Products/Formulator/Widget.py

```diff
@@ -948,54 +986,94 @@ def render(self, field, key, value, REQUEST):
-                                                 field.get_value('date_only') and 'false' or 'true',
-                                                 html_id,
-                                                 use_ampm and '12' or '24',
-                                                 start_day,
-)
+                firstDay: '%s'})},100);</script>""" % (
+                    html_id,
+                    use_ampm and 'I' or 'H',
+                    field.get_value('date_only') and 'false' or 'true',
+                    html_id,
+                    use_ampm and '12' or '24',
+                    start_day,
+                )
         if not field.get_value('date_only'):
             time_result = (field.render_sub_field('hour', hour, REQUEST) +
                            field.get_value('time_separator') +
                            field.render_sub_field('minute', minute, REQUEST))
 
-            hour_id = field.sub_form.get_field('hour').generate_field_html_id("subfield_" + field.id + "_hour")
-            minute_id = field.sub_form.get_field('minute').generate_field_html_id("subfield_" + field.id + "_minute")
-            ampm_id = field.sub_form.get_field('ampm').generate_field_html_id("subfield_" + field.id + "_ampm")
+            hour_id = field.sub_form.get_field(
+                'hour').generate_field_html_id(
+                    "subfield_" + field.id + "_hour")
+            minute_id = field.sub_form.get_field(
+                'minute').generate_field_html_id(
+                    "subfield_" + field.id + "_minute")
+            ampm_id = field.sub_form.get_field('ampm').generate_field_html_id(
+                "subfield_" + field.id + "_ampm")
             if use_ampm:
                 time_result += '&nbsp;' + field.render_sub_field('ampm',
-                                                            ampm, REQUEST)
+                                                                 ampm, REQUEST)
             calendar_popup += calendar_picker and render_element(
-                                'input',
-                                type='hidden',
-                                id=html_id + '_hiddeninput',
-                                onchange='var pattern = /(\d{4})\/(\d{2})\/(\d{2}) (\d{2}):(\d{2}) (am|pm)/; if (pattern.exec(this.value)) { document.getElementById("' + year_id + '").value = RegExp.$1; document.getElementById("' + month_id + '").value = RegExp.$2; ' + select_day + ' document.getElementById("' + hour_id + '").value = RegExp.$4; document.getElementById("' + minute_id + '").value = RegExp.$5; ' + str(use_ampm and 'document.getElementById("' + ampm_id + '").value = RegExp.$6;' or '') + ' }') or ''
-            return date_result + '&nbsp;&nbsp;&nbsp;' + time_result + calendar_popup
+                'input',
+                type='hidden',
+                id=html_id +
+                '_hiddeninput',
+                onchange=r'var pattern = /(\d{4})\/(\d{2})\/(\d{2})'
```

Comment:
This is really something, where a template or string `format()` would be really helpful, as cutting the JS code in this way makes it not readable.

Reply:
I made it more readable.